decorators/timer.py

```diff
@@ -0,0 +1,32 @@
+# ------------------------------------------------------------------------------------
+# Tutorial: Implementation of decorator in Python.
+# ------------------------------------------------------------------------------------
+
+import time
+
+# defining the decorator timer
+def timer(func):
+    # wrapper function taking arbitary arguments
+    def wrapper(*args, **kwargs):
+        # storing the current time.
+        start_time = time.time()
+        # calling func passed in the timer function.
+        func(*args, **kwargs)
+        # displaying the time it takes to completely run the func.
+        print("Function took : ", time.time() - start_time, "seconds.")
+    # returning the wrapper function.
+    return wrapper
+
+# using decorator timer
+@timer
+# defining a function to decorate with timer.
+def run():
+    for i in range(10):
+        print(i)
+
+# calling the function which is decorated with timer.
+run()
+
+# ------------------------------------------------------------------------------------
+# Challenge: Create a decorator to wrap a function which logs the function call with the timestamp when it is called.
```

Comment:
I'm not sure I understand the challenge, what did you mean ? IMO something easier will be better for a first decorator like print something before and after the function or similar.

Reply:
Challenge Updated. Thanks!!